src/WSRequestHandler_Scenes.cpp

```diff
@@ -148,3 +148,119 @@ RpcResponse WSRequestHandler::ReorderSceneItems(const RpcRequest& request) {
 
 	return request.success();
 }
+
+/**
+ * Set a scene to use a specific transition override.
+ *
+ * @param {String} `sceneName` Name of the scene to switch to.
+ * @param {String} `transitionName` Name of the transition to use.
+ * @param {int (Optional)} `transitionDuration` Duration in milliseconds of the transition if transition is not fixed. Defaults 300 if there is no current override and this value is not given.
+ *
+ * @api requests
+ * @name SetSceneTransitionOverride
+ * @category scenes
+ * @since 4.9.0
+ */
+RpcResponse WSRequestHandler::SetSceneTransitionOverride(const RpcRequest& request) {
+	if (!request.hasField("sceneName") || !request.hasField("transitionName")) {
+		return request.failed("missing request parameters");
+	}
+
+	QString sceneName = obs_data_get_string(request.parameters(), "sceneName");
+	OBSSourceAutoRelease source = obs_get_source_by_name(sceneName.toUtf8());
+	if (!source) {
+		return request.failed("requested scene does not exist");
+	} else if (!(obs_source_get_type(source) == OBS_SOURCE_TYPE_SCENE)) {
+		return request.failed("requested scene is invalid");
+	}
+	
+	QString transitionName = obs_data_get_string(request.parameters(), "transitionName");
+	if (!Utils::GetTransitionFromName(transitionName)) {
+		return request.failed("requested transition does not exist");
+	}
+
+	obs_data_t *sourceData = obs_source_get_private_settings(source);
```

Comment:
Never use refcounted pointers directly. For `obs_data_t` pointers, use either one of these containers:
- `OBSDataAutoRelease` for release-only references (when the function providing it internally calls `obs_data_addref`)
- `OBSData` when the reference count isn't automatically incremented by the providing function.

Reply:
`obs_source_get_private_settings` calls `obs_data_addref` internally: https://github.com/obsproject/obs-studio/blob/665180350dc8322346681054bace36d8bf5d1967/libobs/obs-source.c#L4734-L4741

Reply:
Oh and if it's not used before line 191 you can probably put `sourceData` closer to it